internal/contracts/manager.go

```diff
@@ -106,7 +107,12 @@ func NewContractManager(ctx context.Context, ns string, di database.Plugin, bi b
 		core.OpTypeBlockchainContractDeploy,
 	})
 
-	return cm, nil
+	// Validate all our listeners exist on startup - consistent with the multi-party manager.
+	// This means if EVMConnect is cleared, simply a restart of the namespace on core will
+	// cause recreation of all the listeners (noting that listeners that were specified to start
+	// from latest, will start from the new latest rather than replaying from the block they
+	// started from before they were deleted).
```

Comment:
I wonder if, rather than storing "latest" we should resolve what the latest block number is at the time, then store that? This is out of scope for this PR, but may be worth thinking about for better determinism.